InfiniteRecharge/src/main/java/frc/robot/subsystem/drive/DriveSubsystem.java

```diff
@@ -631,7 +631,10 @@ public double getRightVelocity_mps() {
     }
 
 	public Trajectory getFirstPickupTrajectory() {
-        return pickupTrajectoryChooser.getSelected().getFirstPickupTrajectory();
+        Trajectory traj = pickupTrajectoryChooser.getSelected().getFirstPickupTrajectory();
```

Comment:
It’s generally bad form for a getter function to have side effect.

Reply:
bUt It WoRkS

Reply:
How about doing it in autonomous init?

```java
        // reset nav system, turn on LL leds
        navigationSubsystem.reset();
        navigationSubsystem.setInitialPose(driveSubsystem.getFirstPickupTrajectory().getInitialPose());
```

Reply:
or maybe make a separate driveSubsystem method for getInitailPose() that just returns `driveSubsystem.getFirstPickupTrajectory().getInitialPose()`